File: tests/fixtures/openai-realtime/openai-realtime-reconnect-on-drop-broken/01-log-only.ts
import WebSocket from 'ws';

function setup(logger: any) {
  const url = 'wss://api.openai.com/v1/realtime?model=gpt-realtime';
  const callerSocket = new WebSocket(url, { headers: {} });

  callerSocket.on('close', () => {
    logger.info('Caller webSocket connection to OpenAI is closed now.');
  });
}

File: tests/fixtures/openai-realtime/openai-realtime-reconnect-on-drop-broken/01_log_only.py
import logging

import websockets
from websockets.exceptions import ConnectionClosed

WS_URL = "wss://api.openai.com/v1/realtime?model=gpt-realtime"
logger = logging.getLogger(__name__)


async def stream():
    try:
        async with websockets.connect(WS_URL) as ws:
            async for message in ws:
                handle_message(message)
    except ConnectionClosed as exc:
        logger.error("realtime socket closed: %s", exc)
